### PR TITLE
Uwp attachments

### PR DESCRIPTION
**Pull request by codergrl**
@nCastle1 please review
- wired up deleting attachments
- wired up adding new attachments from file and from camera

### CODE REVIEW

File: src/DataCollection.UWP/Views/IdentifiedFeaturePopup.xaml.cs [hunk 1]
```diff
@@ -74,5 +77,21 @@ private async void OnPropertyChanged([CallerMemberName] string propertyName = nu
         }
 
         public event PropertyChangedEventHandler PropertyChanged;
+
+        /// <summary>
+        /// Event handler for user selecting to add a new attachment by capturing new media
+        /// </summary>
+        private async void CaptureMediaButton_Click(object sender, Windows.UI.Xaml.RoutedEventArgs e)
+        {
+            _identifiedFeatureViewModel.AttachmentsViewModel.NewAttachmentFile = await MediaHelper.RecordMediaAsync();
```
Comment (nCastle1) [inline src/DataCollection.UWP/Views/IdentifiedFeaturePopup.xaml.cs]: I would check for exceptions here. If the user uses 'Esc' to close the file name dialog, the file gets saved as `.jpg`; doing that more than once throws an exception due to the duplicate name.

Comment (nCastle1) [reply src/DataCollection.UWP/Views/IdentifiedFeaturePopup.xaml.cs]: Or just doing multiple attachments without changing the name will also throw an exception

File: src/DataCollection.UWP/Views/OriginRelatedRecordPopup.xaml.cs [hunk 2]
```diff
@@ -74,5 +75,21 @@ private async void OnPropertyChanged([CallerMemberName] string propertyName = nu
         }
 
         public event PropertyChangedEventHandler PropertyChanged;
+
+        /// <summary>
+        /// Event handler for user selecting to add a new attachment by capturing new media
+        /// </summary>
+        private async void CaptureMediaButton_Click(object sender, Windows.UI.Xaml.RoutedEventArgs e)
```
Comment (nCastle1) [inline src/DataCollection.UWP/Views/OriginRelatedRecordPopup.xaml.cs:98]: Could these be refactored to be commands instead of using click event handlers?

**Review by nCastle1 — CHANGES_REQUESTED** [review]
Looks good!

One thing, I noticed a crash when attempting to open a recently uploaded (within the session) attachment - filename is null and that causes an exception when creating the FileInfo in *AttachmentsViewModel.cs:161*. 

Only other change I would suggest is clarifying the language used in the attachment deletion warning. It looks like the changes are only persisted after the user explicitly saves, but the warning suggests that as soon as the user hits delete, the attachment is permanently gone.

**Comment by mikewilburn** [comment]
@nCastle1 
> Only other change I would suggest is clarifying the language used in the attachment deletion warning. It looks like the changes are only persisted after the user explicitly saves, but the warning suggests that as soon as the user hits delete, the attachment is permanently gone.

I think the plan was to update this message with something to the effect of "deleted attachments will be permanently gone once the edits are saved" but without using jargon that requires the user be aware of the fact that they're in an active 'edit' session. One possibility to ease this was by referencing the tooltip text of the button with the save icon, but that would also require adding that tooltip text first. Sounds like that didn't make it in here.

File: src/DataCollection.UWP/Views/OriginRelatedRecordPopup.xaml.cs (hunk 2 above)
Comment (nCastle1) [reply src/DataCollection.UWP/Views/OriginRelatedRecordPopup.xaml.cs:98]: Looking at this closer, I see why maybe it isn't as straightforward to do this - the behavior and expectations vary significantly between platforms, enough that is probably doesn't make sense to move that to shared code.

File: src/DataCollection.Shared/ViewModels/AttachmentsViewModel.cs [hunk 3]
```diff
@@ -267,5 +322,52 @@ private async void AddNewAttachment(string filePath)
...
+            using (var stream = await file.OpenReadAsync())
+            {
+                var fileBytes = new byte[stream.Size];
+                using (var reader = new DataReader(stream))
+                {
+                    await reader.LoadAsync((uint)stream.Size);
+                    reader.ReadBytes(fileBytes);
+                }
+                return fileBytes;
+            }
```
Comment (dotMorten) [inline src/DataCollection.Shared/ViewModels/AttachmentsViewModel.cs:380]: Don't read files into byte-arrays. Large files could cause out-of-memory.

File: src/DataCollection.Shared/ViewModels/AttachmentsViewModel.cs [hunk 4]
```diff
@@ -267,5 +322,52 @@ private async void AddNewAttachment(string filePath)
             await stagedAttachment.LoadAsync(newAttachment);
             Attachments.Add(stagedAttachment);
         }
+
+#if NETFX_CORE
+        // HACK: works around issues with attachment management in the UWP sandbox. See OpenAttachment above for details.
```
Comment (dotMorten) [inline src/DataCollection.Shared/ViewModels/AttachmentsViewModel.cs]: It's not a hack nor a workaround. It's dealing with the sandbox.

File: src/DataCollection.UWP/DataCollection.UWP.csproj [hunk 5]
```diff
@@ -11,13 +11,20 @@
     <AssemblyName>DataCollection.UWP</AssemblyName>
     <DefaultLanguage>en-US</DefaultLanguage>
     <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
-    <TargetPlatformVersion Condition=" '$(TargetPlatformVersion)' == '' ">10.0.16299.0</TargetPlatformVersion>
+    <TargetPlatformVersion Condition=" '$(TargetPlatformVersion)' == '' ">10.0.17763.0</TargetPlatformVersion>
```
Comment (dotMorten) [inline src/DataCollection.UWP/DataCollection.UWP.csproj]: Why this? If you rely on 17763, make sure those APIs you need for that are guarded against if running on 16299

File: src/DataCollection.UWP/DataCollection.UWP.csproj [hunk 6]
```diff
@@ -11,13 +11,20 @@
...
     <TargetPlatformMinVersion>10.0.16299.0</TargetPlatformMinVersion>
     <MinimumVisualStudioVersion>14</MinimumVisualStudioVersion>
     <FileAlignment>512</FileAlignment>
     <ProjectTypeGuids>{A5A43C5B-DE2A-4C0C-9213-0A381AF9435A};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
     <WindowsXamlEnableOverview>true</WindowsXamlEnableOverview>
     <PackageCertificateKeyFile>DataCollection.UWP_TemporaryKey.pfx</PackageCertificateKeyFile>
+    <GenerateAppInstallerFile>False</GenerateAppInstallerFile>
+    <AppxAutoIncrementPackageRevision>True</AppxAutoIncrementPackageRevision>
+    <AppxBundle>Always</AppxBundle>
+    <AppxBundlePlatforms>x86|x64|arm</AppxBundlePlatforms>
```
Comment (dotMorten) [inline src/DataCollection.UWP/DataCollection.UWP.csproj]: arm64 missing

File: src/DataCollection.UWP/Helpers/MediaHelper.cs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+namespace Esri.ArcGISRuntime.ExampleApps.DataCollection.UWP.Helpers
+{
+    /// <summary>
+    /// Helper class to process media inputs from both camera and file browsing
+    /// </summary>
+    class MediaHelper
+    {
+        /// <summary>
+        /// Calls camera API to capute photo pr video
```
Comment (dotMorten) [inline src/DataCollection.UWP/Helpers/MediaHelper.cs]: nit: typo :)

File: src/DataCollection.UWP/Views/OriginRelatedRecordPopup.xaml.cs [hunk 8]
```diff
@@ -74,5 +75,21 @@ private async void OnPropertyChanged([CallerMemberName] string propertyName = nu
...
+        {
+            _identifiedFeatureViewModel.SelectedOriginRelationship.AttachmentsViewModel.NewAttachmentFile = await MediaHelper.RecordMediaAsync();
```
Comment (dotMorten) [inline src/DataCollection.UWP/Views/OriginRelatedRecordPopup.xaml.cs]: `async void` methods should have try/catch on any awaited methods.

**Review by dotMorten — CHANGES_REQUESTED** [review]
Generally looking good, but see comments

**Comment by mikewilburn** [comment]
One behavior I'm noticing so far is that when canceling the download of an offline map, it seems to always kill the app. Commenting here in case that might have been introduced by this PR.

**Pull request merged**